### ISSUE TITLE
LaserOdometry::hasNewData() and LaserMapping::hasNewData() depends on your computer's speed

### ISSUE THREAD

**Issue by claydergc** [laboshinl/loam_velodyne#103]
Hello,

I was reviewing the code and in the class LaserOdometry and LaserMapping there are some thresholds that depend on the difference on how fast a pair of topics are published. An example is this line in LaserOdometry::hasNewData() function:

`fabs((_timeCornerPointsSharp - _timeSurfPointsLessFlat).toSec()) < 0.005`

This line depends on the difference in times when topics cornerPointsSharp surfPointsLessFlat were published and this depends on how fast your computer does it. I have been experimenting with these values and sometimes the estimated trajectory can fail.

**Comment by wonder566** [comment]
right, i found this too, when i build the package in debug mode it works only with lower pointcloud publishing speed
